File: SpacifyAPI/SpacifyAPI/Models/Responses/UserResponseForAdmin.cs
using SpacifyAPI.Entities;
using SpacifyAPI.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace SpacifyAPI.Models.Responses
{
    public class UserResponseForAdmin
    {
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public string? Surname { get; set; }
        public string? Email { get; set; }
        public string? Username { get; set; }
        public string? Role { get; set; }

        public int FailedLoginAttempts { get; set; }
        public DateTimeOffset? AccountBlockedUntil { get; set; }

        public bool IsBlocked { get; set; } = false;

        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset? UpdatedAt { get; set; }
        public DateTimeOffset? LastLoginAt { get; set; }

        //public List<WorkstationReservation>? WorkstationReservations { get; set; } = new List<WorkstationReservation>();
        //public List<ConferenceRoomReservation>? ConferenceRoomReservations { get; set; } = new List<ConferenceRoomReservation>();
    }
}

File: SpacifyFrontend/src/app/models/response/user-response-for-admin.ts
import { UserRole } from '../../enums/user-role.enum';

export interface UserResponseForAdmin {
  id: string;
  name: string;
  surname: string;
  email: string;
  username: string;
  role: UserRole;
  failedLoginAttempts: number;
  accountBlockedUntil: Date | null;
  isBlocked: boolean;
  createdAt: Date;
  updatedAt?: Date | null;
  lastLoginAt?: Date | null;
}
